[synthetics] Update docs to reflect the changes to `expect` (#3297)
elastic/observability-docs: docs/en/observability/synthetics-create-test.asciidoc

Visit the https://playwright.dev/docs[Playwright documentation] for information.

However, not all Playwright functionality should be used with Elastic Synthetics.
In some cases, there are alternatives to Playwright functionality built into the
Elastic Synthetics library. These alternatives are designed to work better for
synthetic monitoring. Do _not_ use Playwright syntax to:

* *Make API requests.* Use Elastic Synthetic's `request`
  parameter instead. Read more in <<synthetics-request-param>>.

There is also some Playwright functionality that is not supported out-of-the-box
in Elastic Synthetics including:

* https://playwright.dev/docs/api/class-video[Videos]
* The https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-have-screenshot-1[`toHaveScreenshot`] and https://playwright.dev/docs/api/class-snapshotassertions[`toMatchSnapshot`] assertions

[discrete]
[[synthetics-make-assertions]]
= Make assertions

A more complex `step` might wait for a page element to be selected
and then make sure that it matches an expected value.

> Elastic Synthetics uses `@playwright/test`'s `expect` function to make assertions and

Reviewer: Would call this unsupported one as snapshot based assertions.  `except for the unsupported ones [Snapshot Assertions](https://playwright.dev/docs/api/class-snapshotassertions) toHaveScreenshot and toMatchSnapsho`

supports all https://playwright.dev/docs/test-assertions[Playwright assertions] _except_
`toHaveScreenshot` and `toMatchSnapshot`.

> 

Author: @vigneshshanmugam I'm not sure I understand your comment. Is this clearer and still correct?

>Elastic Synthetics uses `@playwright/test`'s `expect` function to make assertions and supports most [Playwright assertions](https://playwright.dev/docs/test-assertions). Elastic Synthetics does _not_ support [`toHaveScreenshot`](https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-have-screenshot-1) or any [Snapshot Assertions](https://playwright.dev/docs/api/class-snapshotassertions).

Reviewer: Sorry should have been eloborate. yeah I was just asking if we can link in to that Snapshot based assertions. Looks good to me.

Author: ```suggestion
Elastic Synthetics uses `@playwright/test`'s `expect` function to make assertions
and supports most https://playwright.dev/docs/test-assertions[Playwright assertions].
Elastic Synthetics does _not_ support https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-have-screenshot-1[`toHaveScreenshot`]
or any https://playwright.dev/docs/api/class-snapshotassertions[Snapshot Assertions].

```
